apps/condo/domains/organization/integrations/sbbol/sync/requestTransactions.js

```diff
@@ -52,22 +52,36 @@ async function requestTransactionsForDate ({ userId, bankAccounts, context, stat
         let page = 1,
             doNextRequest = true,
             timeout = 1000,
-            failedReq = 0
+            failedReq = 0,
+            summary
 
         do {
             let response = await fintechApi.getStatementTransactions(bankAccount.number, statementDate, page)
+            summary = await fintechApi.getStatementSummary(
+                bankAccount.number,
+                statementDate,
+            )
 
-            while (!isTransactionsReceived(response)) {
+            while (!isResponceSuccessfull(response) || !isResponceSuccessfull(summary)) {
                 if (failedReq > 5) {
                     break
                 }
                 await new Promise( (resolve) => {
                     setTimeout(async () => {
-                        response = await fintechApi.getStatementTransactions(
-                            bankAccount.number,
-                            statementDate,
-                            page
-                        )
+                        if (!isResponceSuccessfull(response)) {
+                            response = await fintechApi.getStatementTransactions(
+                                bankAccount.number,
+                                statementDate,
+                                page
+                            )
+                        }
+
+                        if (!isResponceSuccessfull(summary)) {
+                            summary = await fintechApi.getStatementSummary(
+                                bankAccount.number,
+                                statementDate,
+                            )
+                        }
```

Comment:
You have a call of this method before this "while" loop.
For what purpose you have this extra call?

Reply:
when I send a request to receive an account statement, I may get back an error saying that the statement is being generated. Therefore, I first request it and, if I don’t receive it, I go to request it in a loop with a limit of 5 reps and a timeout that doubles after each fail

Reply:
Could you please create a comment with all what you answered here, because it's not obvious from the fist look.